packages/app/src/components/views/publication/PublicationPostView.tsx

```diff
@@ -78,12 +79,15 @@ export const PublicationPostView: React.FC<PublicationPostViewProps> = ({ update
               <PublicationPostContainer container gap={3} alignItems={"center"}>
                 <Grid item>
                   {!editingPublication && (
-                    <Avatar
-                      sx={{ width: 160, height: 160 }}
-                      src={publication.image ? `${IPFS_GATEWAY}/${publication.image}` : ""}
-                    >
-                      {" "}
-                    </Avatar>
+                    <DeterministicAvatar publicationId={publicationId} width={160} height={160} />
```

Comment:
@auryn-macmillan temporarily I am just including the generated canvas here where the avatar would go. I think it would be best to somehow upload this generated canvas to ipfs as a flat file (png or jpg). This should be possible, right? That way the image could just be included in the `publication` data.

Reply:
Oh, this is an interesting idea.

I wonder if we could actually write the image generation directly in the subgraph and then have the subgraph just store the image directly?